habitat-lab/habitat/tasks/rearrange/rearrange_task.py

```diff
@@ -68,7 +68,8 @@ def __init__(
         **kwargs,
     ) -> None:
         self.n_objs = len(dataset.episodes[0].targets)
-
+        self.n_objs = np.max([self.n_objs, kwargs['config'].max_objects])
```

Comment:
@ASzot this is change (2). `kwargs['config'].max_objects ` is 0, unless explicitly specified in the task config.